### PR TITLE
Added from_sorted_iter to fix #69

### LINKED ISSUE #69: Optimized Insert / Extend for sorted data
In my current use case, 65% of the time is spent doing the binary search inside of the `insert` method of `Store` but all the data are inserted in order.

Therefore, an optimized method for sorted values might speed-up this case.

I'm happy to help and I'm currently looking at the code to see If I'm able to fix it and submit a pull request.

### PR DESCRIPTION
**Pull request by zommiommy**
As explained in #69 in my current use case ~65% of the time is spent doing binary searches when inserting values in the bitmap.

I always have sorted values, therefore the binary search can be skipped.
For this reason, I implemented the following methods for both bitmaps.

- `push` add the new "max" to the set.
- `append` take an iterator of sorted values (all bigger than the max of the set) and add them to the set.
- `from_sorted_iter` creates a set from an iterator of sorted values.

I tried choosing self-explaining names for the methods but I'm not married to them and open to changes.

In order to assure that the values are actually sorted I inserted two asserts in the push method.

### CODE REVIEW

File: src/bitmap/store.rs
Comment (Kerollmops) [inline src/bitmap/store.rs:49]:
```suggestion
                if vec.last().map_or(true, |x| x < &index) {
```

Maybe this could work and avoid an unwrap.

File: src/bitmap/inherent.rs
Comment (Kerollmops) [inline src/bitmap/inherent.rs:61]:
```suggestion
    pub fn push_back(&mut self, value: u32) {
```

I am just proposing that as it is a little bit more explicit/obvious, what do you think?

File: src/bitmap/store.rs
Comment (zommiommy) [reply src/bitmap/store.rs:49]: looks good to me!

File: src/bitmap/inherent.rs
Comment (zommiommy) [reply src/bitmap/inherent.rs:61]: I used push because to be consistent with Vectors, but If you think this is better go for it!

Comment (Kerollmops) [reply src/bitmap/inherent.rs:61]: Not sure it is better, can the pushed integer be also equal to the last integer in the collection, too?

Comment (zommiommy) [reply src/bitmap/inherent.rs:61]: Thank you for your note, you are right, I have just fixed that case in the [last commit](https://github.com/Nemo157/roaring-rs/pull/70/commits/d9f0de7f71e8e090648111ed8b3b7d4f64ded610)

File: src/bitmap/inherent.rs
Comment (Kerollmops) [inline src/bitmap/inherent.rs:61]: What does the boolean value means? If this method is only able to insert at the end an integer that do not already exist it should always return `true`, or does it return `false` when the operation is not a success?

You could maybe add a note about the returned value? :)

Comment (zommiommy) [reply src/bitmap/inherent.rs:61]: Sure! It's like the in the insert method, basically it return false if it's a duplicate of the last value. Sorry for the mess, I'll have dinner and this evening or in the morning I'll extend the documentation.

Comment (zommiommy) [reply src/bitmap/inherent.rs]: After sleeping on it, I think that, as you pointed out, the bool return might be confusing so I removed it.

File: src/bitmap/inherent.rs
Comment (Kerollmops) [inline src/bitmap/inherent.rs:48]:
```suggestion
    /// The value **must** be bigger or equal to the maximum value in the set.
```

Comment (Kerollmops) [reply src/bitmap/inherent.rs]: aha ! No I wanted to chnage the text, you are too fast!

> The value **must** be greater or equal than the maximum value in the set.

Comment (Kerollmops) [inline src/bitmap/inherent.rs]:
```suggestion
    /// The value **must** be greater or equal than the maximum value in the set.
```

File: src/bitmap/store.rs
Comment (Kerollmops) [inline src/bitmap/store.rs:49]: Just to have a better rendering of the doc :)

```suggestion
    /// Push the value that must be the new max of the set.
    ///
    /// This function returns whether the value is equal to the
    /// last max. This infomation is needed to correctly update the
    /// length of the container.
```

**Review by Kerollmops — CHANGES_REQUESTED** [review]
Thank you very much for this PR, could you just change this little things before I merge?

**Comment by Kerollmops** [comment]
Hey @zommiommy,

I am the new maintainer of this repository and wanted to make sure that merging or testing this PR works correctly, I just updated the CI and the bors settings, so could it be possible that you rebase on master please? (don't forget to pull master before rebasing 😄)

**Comment by zommiommy** [comment]
Hi, thanks a lot! In a couple of hours max I'll be able to do work on it. Could you explain to me what rebasing on master means? I'm pretty new to pull-requests 😅 Thanks for the patience.

**Comment by Kerollmops** [comment]
No worries, take your times there is no rush at all ;)

Rebasing means pulling the master branch of this repository and remapping your list of commits that are part of this PR on top of master, this way your PR will contains the new CI and bors settings.

[Here is a simple tutorial](https://gist.github.com/curquiza/5f7ce615f85331f083cd467fc4e19398), but don't hesitate to @ me if you need help.

**Comment by zommiommy** [comment]
Is this okei?

Moreover, I can't seem to accept your suggestion for the `inherent.rs` file

File: src/bitmap/inherent.rs
Comment (Kerollmops) [inline src/bitmap/inherent.rs:47]:
```suggestion
    /// The value **must** be greater or equal to the maximum value in the set.
```

File: src/bitmap/iter.rs
Comment (Kerollmops) [inline src/bitmap/iter.rs:141]:
```suggestion
    /// Create the set from a sorted iterator. Values **must** be sorted.
```

File: src/treemap/inherent.rs
Comment (Kerollmops) [inline src/treemap/inherent.rs:45]:
```suggestion
    /// The value **must** be greater or equal to the maximum value in the set.
```

Am I right on this? Can values be equal to the max even for treemaps?

File: src/treemap/iter.rs
Comment (Kerollmops) [inline src/treemap/iter.rs:226]:
```suggestion
    /// Create the set from a sorted iterator. Values **must** be sorted.
```

Am I right here too? Can values be also equal?

File: src/treemap/iter.rs
Comment (Kerollmops) [inline src/treemap/iter.rs:246]:
```suggestion
    /// All value of the iterator **must** be greater or equal than the max element
```

File: tests/push.rs
Comment (Kerollmops) [inline tests/push.rs:37]: Yeah, it seems that I am right!

**Review by Kerollmops — CHANGES_REQUESTED** [review]
I just add a bunch of updates to the method documentations. If the github "accept suggestion" feature doesn't work, could you copy/paste my propositions? :)

**Review by Kerollmops — APPROVED** [review]
Thank you very much, I am merging this PR! 🥳

bors r+

File: tests/push.rs
Comment (zommiommy) [reply tests/push.rs:37]: Yeah!

**Comment by zommiommy** [comment]
Thanks to you for your patience!

**Comment by Kerollmops** [comment]
I will probably bump and release a next version when #68 is merged.

**Pull request merged**
